src/utilities/phone/index.ts

```diff
@@ -18,20 +18,56 @@ export function isValidDDD(phone: string): boolean {
   return VALID_AREA_CODES.includes(Number(phone.substr(0, 2)));
 }
 
+export function isValidMobilePhoneLength(phone: string): boolean {
+  return phone.length === PHONE_MAX_LENGTH;
+}
+
+export function isValidLandlinePhoneLength(phone: string): boolean {
+  return phone.length >= PHONE_MIN_LENGTH && phone.length < PHONE_MAX_LENGTH;
+}
+
 export function isValidLength(phone: string): boolean {
-  return phone.length >= PHONE_MIN_LENGTH && phone.length <= PHONE_MAX_LENGTH;
+  return isValidLandlinePhoneLength(phone) || isValidMobilePhoneLength(phone);
+}
+
+export function isValidMobilePhoneFirstNumber(phone: string): boolean {
+  return MOBILE_VALID_FIRST_NUMBERS.includes(Number(phone.charAt(2)));
+}
+
+export function isValidLandlinePhoneFirstNumber(phone: string): boolean {
+  return LANDLINE_VALID_FIRST_NUMBERS.includes(Number(phone.charAt(2)));
 }
 
 export function isValidFirstNumber(phone: string): boolean {
   return phone.length === PHONE_MIN_LENGTH
-    ? LANDLINE_VALID_FIRST_NUMBERS.includes(Number(phone.charAt(2)))
-    : MOBILE_VALID_FIRST_NUMBERS.includes(Number(phone.charAt(2)));
+    ? isValidLandlinePhoneFirstNumber(phone)
+    : isValidMobilePhoneFirstNumber(phone);
+}
+
+function tryPhoneDigits(phone: string): { isValidDigits: boolean; digits: string } {
+  return { isValidDigits: !!phone && typeof phone === 'string', digits: onlyNumbers(phone) };
+}
+
+export function isValidMobilePhone(phone: string): boolean {
+  const { isValidDigits, digits } = tryPhoneDigits(phone);
+
+  if (!isValidDigits) return false;
+
+  return isValidMobilePhoneLength(digits) && isValidMobilePhoneFirstNumber(digits) && isValidDDD(digits);
+}
+
+export function isValidLandlinePhone(phone: string): boolean {
+  const { isValidDigits, digits } = tryPhoneDigits(phone);
+
+  if (!isValidDigits) return false;
+
+  return isValidLandlinePhoneLength(digits) && isValidLandlinePhoneFirstNumber(digits) && isValidDDD(digits);
 }
 
 export function isValid(phone: string): boolean {
-  if (!phone || typeof phone !== 'string') return false;
+  const { isValidDigits, digits } = tryPhoneDigits(phone);
```

Comment:
I think that you could rewrite to something like this:

```
export function isValid(phone: string): boolean {
  return isValidLandlinePhone(phone) || isValidMobilePhone(phone);
}
```

What do you think about it?

Reply:
Hm.... It was to avoid to call `tryPhoneDigits()` two times, however, I'm going to change as suggested.